pkg/provider/aws/plugin.go

```diff
@@ -52,21 +52,50 @@ func (a *AWS) CopyConstructEdgesToDag(result *core.ConstructGraph, dag *core.Res
 		data := knowledgebase.EdgeData{AppName: a.Config.AppName, Source: sourceResource, Destination: targetResource}
 		switch construct := dep.Source.(type) {
 		case *core.ExecutionUnit:
-			switch dep.Destination.(type) {
+			if a.Config.GetExecutionUnit(construct.ID).Type == kubernetes.KubernetesType {
+				data.SourceRef = construct.AnnotationKey
+			}
+			switch dest := dep.Destination.(type) {
 			case *core.Orm:
 				data.Constraint = knowledgebase.EdgeConstraint{
 					NodeMustExist: []core.Resource{&resources.RdsProxy{}},
 				}
+				// Because we dont understand resource types, our edges cant depict what is truly valid and what isnt yet. Example
```

Comment:
> Because we dont understand resource types

Not sure what this means - aren't resource types all over the knowledge base and resource graph?

Reply:
sorry, because we dont understand that helm chart and lambda are both compute resources is closer to what it means. Our KB should (and hopefully will soon) have the knowledge that it doesnt need to put compute between compute and data but right now has no idea because we dont classify any resource types so lambda -> helm chart -> rds is a valid path to it

Reply:
yeah i think the next designs which we will hopefully be picking up next week will address this